### DISCUSSION TITLE
Check if file is encrypted with openpgp

### DISCUSSION THREAD

**Discussion by Tzvetelin88** [openpgpjs/openpgpjs#1394] [Q&A]
Is there a way to check if file is encrypted with openpgp without private/public key?
Idea is to check file signature and if encrypted with openpgp to ask for encryption key (this is web implemented in web application).

**Comment by larabr** [comment]
Hi @Tzvetelin88, if the encrypted data is in string format (armored), then you can just check that it starts with `-----BEGIN PGP MESSAGE-----`. If you are working over a binary input, I'd just try to parse it as a message with `openpgp.readMessage({ binaryMessage: ... })`. If it succeeds, then it must be OpenPGP data.
